peach_tuntiwong/week07/day31/main.js

```diff
@@ -80,7 +87,12 @@ $(document).ready(function(){
     xhr.onreadystatechange = function(){
       if (xhr.readyState == 4) {
         var json_response = JSON.parse(xhr.responseText);
-        var plot = $("#"+ json_response.imdbID).append($("<p>").html(json_response.Plot));
+
+        // Check if the movie_div doesn't have plot displayed
+        if($("#"+ json_response.imdbID).children().is("p") == false){
+          var plot = $("#"+ json_response.imdbID).append($("<p>").html(json_response.Plot));
+        }
```

Comment:
Nice work... but maybe you should test if the plot is already displayed _before_ you make the whole AJAX request, to save you making it if you don't need to. In other words:
```
if($("#"+ json_response.imdbID).children().is("p") == false){
  xhr.open("GET", "http://www.omdbapi.com/?i=" + id);
  xhr.send();
}
```

You also might be interested in the `.has()` jQuery method, which would make your test a bit simpler, because it searches all the children for any of the specified elements:
```
if( $("#"+ json_response.imdbID).has('p').length === 0 ){
```